### ISSUE TITLE
Rename to rails-template?

### ISSUE THREAD

**Issue by croaky** [thoughtbot/suspenders#827]
We have these other templates for languages and frameworks:

* https://github.com/thoughtbot/android-template
* https://github.com/thoughtbot/django-template
* https://github.com/thoughtbot/ios-template
* https://github.com/thoughtbot/react-native-template

Opened https://github.com/thoughtbot/laces/pull/10 to rename Laces to Go template.

**Comment by croaky** [comment]
Fun fact: this was the original name of this repo back pre-GitHub 2007-ish when it was in a thoughtbot Subversion repo.

**Comment by tute** [comment]
Note that a Rails Template is a preexisting concept in Rails: http://guides.rubyonrails.org/rails_application_templates.html

**Comment by jessieay** [comment]
I like the name suspenders! I feel like template makes sense, but I think suspenders is more fun 👖

**Comment by mike-burns** [comment]
Our hope (my hope? someone's hope) is to move suspenders away from being a command-line tool and toward being a set of rake tasks. Does the idea of suspenders-as-a-template remain consistent with the suspenders-as-rake-tasks perspective?

**Comment by toobulkeh** [comment]
From a high-level architecture view, suspenders is a template with defaults. Running them as rake tasks after the defaults have already been setup from `rails new` is... misleading at best?

I don't think launching it from a rake task makes any more sense. The only advantage I immediately see is being able to know which version of suspenders (defaults) the project started with. But that's best held by a date/time in git -- not an artifact in a codebase -- unless "upgrading" is supportable in suspenders, which is more of a process for an Engine, not a template.

**Comment by croaky** [comment]
Suspenders has staying power.

**Comment by toobulkeh** [comment]
@mike-burns not sure if this is the right forum for this -- but one thing we've been discussing is how nice it would be to keep all of our projects up to date with core suspenders changes.

Is there such a thing as a framework-wrapper that could be versioned? Would this just be a CLI gem? Could Rails be extended as another framework?

**Comment by mike-burns** [comment]
@toobulkeh for more history here, see #511, #765, #766, #767, and #855.

**Comment by toobulkeh** [comment]
TL;DR: No - Suspenders is a generator only. OK.

**Comment by nickcharlton** [comment]
> but one thing we've been discussing is how nice it would be to keep all of our projects up to date with core suspenders changes.

We've spoken a bit about this in London, too (and I presume it's come up elsewhere outside of the issues/PRs @mike-burns linked to). I don't know too much about it, but there _is_ an experiment kicking about looking at doing exactly what you'd quoted. As far as I understand it, it's not the easiest thing to figure out but we're looking into it.

**Comment by mike-burns** [comment]
There was also the [trout](https://github.com/thoughtbot/trout) era, as can be seen in fbf39d879092b6b389a2fe4ec6c943e632015d3b, 3c11591f0defcb2281bc9519d78964b461596b2b, 1b4514d01be7f62d2fe73030596a65031a28fe48, a92abc81ed032e7e968fecd634555531d86502fd, 7b9bffcb094a2d7248baf4eaa87f9f1c8fae9df7, and removed in 2b2c6ec78dc4a7a85966db3adc9506db6387acdd, 385dd5b80e5fa73cf118f64105df2405a2351e24, 470e642e943f51c18933ce73edb49161f73b34e9.

**Comment by toobulkeh** [comment]
I knew I wasn't the only one -- thanks for the references, good learnings. @nickcharlton I'd love to see the fruit of those labors -- or help with the laboring if you need a pair.
